be/src/exec/vectorized/analytor.cpp

```diff
@@ -280,24 +295,34 @@ Status Analytor::close(RuntimeState* state) {
 
     _is_closed = true;
 
-    for (auto* ctx : _agg_fn_ctxs) {
-        if (ctx != nullptr && ctx->impl()) {
-            ctx->impl()->close();
+    auto agg_close = [this, state]() {
+        for (auto* ctx : _agg_fn_ctxs) {
+            if (ctx != nullptr && ctx->impl()) {
+                ctx->impl()->close();
+            }
         }
-    }
 
-    // Note: we must free agg_states before _mem_pool free_all;
-    _managed_fn_states.clear();
-    _managed_fn_states.shrink_to_fit();
+        // Note: we must free agg_states before _mem_pool free_all;
+        _managed_fn_states.clear();
+        _managed_fn_states.shrink_to_fit();
 
-    if (_mem_pool != nullptr) {
-        _mem_pool->free_all();
-    }
+        if (_mem_pool != nullptr) {
+            _mem_pool->free_all();
+        }
 
-    Expr::close(_order_ctxs, state);
-    Expr::close(_partition_ctxs, state);
-    for (const auto& i : _agg_expr_ctxs) {
-        Expr::close(i, state);
+        Expr::close(_order_ctxs, state);
+        Expr::close(_partition_ctxs, state);
+        for (const auto& i : _agg_expr_ctxs) {
+            Expr::close(i, state);
+        }
+        return Status::OK();
+    };
+
+    if (_has_udaf) {
+        auto promise_st = call_function_in_pthread(state, agg_close);
```

Comment:
`Operator::close()` is always called in `DriverExecutor::worker_thread`, maybe we needn't this? But maybe it is ok, because it is a stronger guarantee that we can call `close()` in bthread in the future.

Reply:
Suppose there is a case like this:
bthread:
    driver->prepare()
      ExprContext[0]->prepare()/open() success.
      ExprContext[1]->prepare()/open() failed.
then we will call Expr::close for all Expr.
      ExprContext[0]->close() will be called